docs/USER_GUIDE.md

```diff
@@ -300,6 +301,57 @@ spec:
 - Consider adding authentication and rate limiting for production use
 - Use HTTPS in production environments
 
+### Serving Workflow via HTTP API
+
+The `maestro serve-workflow` command allows you to expose Maestro workflow as HTTP API endpoints, making them accessible via REST API calls. This is useful for integrating workflow into another workflows, web applications, microservices, or other systems that need to communicate with AI workflow.
+
+#### Basic Usage
+
+```bash
+# Serve a single agent from an agents file
+maestro serve-workflow workflow.yaml
+
+# Serve on a custom port and host
+maestro serve-workflow agents.yaml workflow.yaml --port 8080 --host 0.0.0.0
+```
+
+#### API Endpoints
+
+Once the server is running, the following endpoints are available:
+
+**POST /chat** - Send prompts to the workflow
+```bash
+curl -X POST "http://127.0.0.1:8000/chat" \
+     -H "Content-Type: application/json" \
+     -d '{"prompt": "Hello, how are you?", "stream": false}'
+```
+
+Response:
+```json
+{
+  "response": "Hello! I'm doing well, thank you for asking. How can I help you today?",
```

Comment:
Also worth noting that if run with a workflow this is not what "response" looks like, it's a stringified json object